error MSB1008 when MSBuildBinaryLogSettings.FileName has spaces

### What You Are Seeing?
MSBuild error MSB1008 when using a BinaryLogger and a path with space
### What is Expected?
no error
### What version of Cake are you using?
0.33
### Are you running on a 32 or 64 bit system?
32 bit
### What environment are you running on?  Windows? Linux? Mac?
Windows 10 1903
### Are you running on a CI Server?  If so, which one?
No
### How Did You Get This To Happen? (Steps to Reproduce)
Add a BinaryLogger and specify a path with space.
```csharp
var msBuildSettings = new MSBuildSettings()
{
            ...
            BinaryLogger = new MSBuildBinaryLogSettings
            {
                Enabled = true,
                FileName = "Path with\spaces in between\log.binlog"
            },
            ...
        };
```

### Output Log
![unknown](https://user-images.githubusercontent.com/6222752/58357152-08303e00-7e7a-11e9-8727-07416bdbf437.png)

--- a/src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs
+++ b/src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs
@@ -993,7 +993,7 @@ namespace Cake.Common.Tests.Unit.Tools.MSBuild
                 var result = fixture.Run();
 
                 // Then
-                Assert.Equal("/v:normal /target:Build /bl:mylog.binlog;ProjectImports=ZipFile \"C:/Working/src/Solution.sln\"", result.Args);
+                Assert.Equal("/v:normal /target:Build /bl:\"mylog.binlog\";ProjectImports=ZipFile \"C:/Working/src/Solution.sln\"", result.Args);
             }
 
             [Fact]
@@ -1007,7 +1007,25 @@ namespace Cake.Common.Tests.Unit.Tools.MSBuild
                 var result = fixture.Run();
 
                 // Then
-                Assert.Equal("/v:normal /target:Build /bl:mylog.binlog;ProjectImports=ZipFile \"C:/Working/src/Solution.sln\"", result.Args);
+                Assert.Equal("/v:normal /target:Build /bl:\"mylog.binlog\";ProjectImports=ZipFile \"C:/Working/src/Solution.sln\"", result.Args);
+            }
+
+            [Fact]
+            public void Should_Quote_Binary_Logging_FilePath()
+            {
+                // Given
+                var fixture = new MSBuildRunnerFixture(false, PlatformFamily.Windows);
+                fixture.Settings.BinaryLogger = new MSBuildBinaryLogSettings()
+                {
+                    Enabled = true,
+                    FileName = "C:/Working Directory/src folder/mylog.binlog"
+                };
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("/v:normal /target:Build /bl:\"C:/Working Directory/src folder/mylog.binlog\" \"C:/Working/src/Solution.sln\"", result.Args);
             }
 
             [Theory]

--- a/src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs
+++ b/src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs
@@ -150,7 +150,7 @@ namespace Cake.Common.Tools.MSBuild
                 string binaryOptions = null;
                 if (!string.IsNullOrEmpty(settings.BinaryLogger.FileName))
                 {
-                    binaryOptions = settings.BinaryLogger.FileName;
+                    binaryOptions = settings.BinaryLogger.FileName.Quote();
                 }
 
                 if (settings.BinaryLogger.Imports != MSBuildBinaryLogImports.Unspecified)

